core/src/storage/device.py

```diff
@@ -283,9 +285,20 @@ def set_sd_salt_auth_key(auth_key: Optional[bytes]) -> None:
         return common.delete(_NAMESPACE, _SD_SALT_AUTH_KEY, public=True)
 
 
-def unsafe_prompts_allowed() -> bool:
-    return common.get_bool(_NAMESPACE, _UNSAFE_PROMPTS_ALLOWED)
+# do not use this function directly, see apps.common.safety_checks instead
+def safety_check_level() -> EnumTypeSafetyCheckLevel:
+    level = common.get_uint8(_NAMESPACE, _SAFETY_CHECK_LEVEL)
+    if level is None:
+        level = _DEFAULT_SAFETY_CHECK_LEVEL
+    # SafetyCheckLevel.PromptTemporarily must not be written to storage
+    if level not in (SafetyCheckLevel.Strict, SafetyCheckLevel.PromptAlways):
```

Comment:
I'm wondering if these two checks should be folded into one:
```
if level not in (...):
	return _DEFAULT_SAFETY_CHECK_LEVEL
else:
	return level
```

an invalid value in _getter_ could have gotten there either through storage corruption, or by a future firmware that uses a different set of values. in either case ISTM erroring out is not the right thing to do? while defaulting to STRICT is reasonable.

Reply:
I would agree with @matejcik , the argument about firmware downgrade is persuasive.

Reply:
Fixed.